analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -121,9 +128,37 @@ public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCod
         // the decision is made on based on whether the project contains the analyzer as NuGet).
         if (context.HasMatchingScope(supportedDiagnostics)
             && context.ShouldAnalyzeTree(sourceTree, generatedCodeRecognizer)
+            && context.Options.ParseSonarLintXmlSettings() is { } sonarLintSettings
+            && IsIncluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
+            && !IsExcluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
             && LegacyIsRegisteredActionEnabled(supportedDiagnostics, context.Tree))
         {
             action(context);
         }
     }
+
+    private bool IsIncluded(Compilation compilation, string[] inclusions, string filePath) =>
+        inclusions is null
+        || inclusions.Length == 0
+        || FindOrCreate(compilation, InclusionsCache, inclusions, filePath);
+
+    private bool IsExcluded(Compilation compilation, string[] exclusions, string filePath) =>
+        exclusions is { Length: > 0 } && FindOrCreate(compilation, ExclusionsCache, exclusions, filePath);
+
+    private bool FindOrCreate(Compilation compilation,
+                              ConditionalWeakTable<Compilation, ConcurrentDictionary<string, bool>> compilationCache,
+                              string[] set,
+                              string filePath)
+    {
+        var cache = compilationCache.GetValue(compilation, x => new());
+        if (!cache.TryGetValue(filePath, out var wrapper))
+        {
+            wrapper = set.Any(x => IsMatch(x, filePath));
+            cache[filePath] = wrapper;
+        }
+        return wrapper;
+    }
+
+    private bool IsMatch(string pattern, string filePath) =>
```

Comment:
Is there a need for this method? It is only used in 1 place so you can call the `globPatternMatcher` directly.